app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesViewHolder.kt

```diff
@@ -52,9 +52,15 @@ class PocketStoriesViewHolder(
     override fun Content() {
         val horizontalPadding = dimensionResource(R.dimen.home_item_horizontal_margin)
 
+        val homeScreenReady = components.appStore
+            .observeAsComposableState { state -> state.firstFrameDrawn }.value ?: false
+
         val stories = components.appStore
             .observeAsComposableState { state -> state.pocketStories }.value
 
+        // Perceived performance can be improved by waiting to draw pocket content until after the
+        // first frame is drawn. See https://github.com/mozilla-mobile/fenix/issues/25545 for details.
+        if (!homeScreenReady) return
```

Comment:
same here

Reply:
Here, there's additional logic that records telemetry when `stories` has a value. There can definitely be cases where `stories` is usable but we are still waiting for the homescreen to be ready. I don't think we want to record that telemetry until the stories are actually show, based on the code and comments.